Sources/Support/PaywallExtensions.swift

```diff
@@ -0,0 +1,153 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PaywallExtensions.swift
+//
+//  Created by Nacho Soto on 6/6/23.
+
+import StoreKit
+import SwiftUI
+
+#if swift(>=5.9)
+
+@available(iOS 17.0, macOS 14.0, tvOS 17.0, watchOS 10.0, *)
+extension StoreView {
+
+    /// Creates a view to load a collection of products from the App Store, and merchandise them.
+    /// When the user purchases products through this paywall, the `RevenueCat` SDK will handle
+    /// the result automatically. All you need to do is to dismiss the paywall
+    public init(
+        offering: Offering,
```

Comment:
This is probably slightly more complex but I was thinking, could we provide another override to a view that actually does the `getOfferings` call and displayes the `StoreView`/`SubscriptionStoreView` once those have loaded? The complex think about that is that we would need to write loading and error UI states ourselves (would need design input). Good thing is that it would make it even easier to use RevenueCat (with this you still would need to use `getOfferings` first and pass the current offering). CC @aboedo

Reply:
That's what `CurrentOfferingSubscriptionStoreView` does below 😄

Reply:
🤦 I got confused, but yeah that's what I meant 🙌